### PR TITLE
jquery upgrade main

### PR DESCRIPTION
**Pull request by smithellis**
Many changes to capture jquery deprecations. Still does not resolve [1089,](https://github.com/mozilla/sumo/issues/1089) but didn't want to hold back all these changes as we move towards completion/rejection of the upgrade.

### CODE REVIEW

**Comment by akatsoulas** [comment]
@smithellis is this ready for review?

**Comment by smithellis** [comment]
@akatsoulas Yes, I think so. I was holding out for resolution to https://github.com/mozilla/sumo/issues/1089 but these changes should probably proceed.

**Comment by akatsoulas** [comment]
Could you squash everything in a single commit? Also are all the files in here needed? For example there is a `jquery-migrate.js` and a `jquery-migrate-141.js` file. I haven't gone through the code so maybe both are needed. I can do a first pass once is squashed in a single commit. (you can do a rebase locally, squash all the commits into a single one and (force) push back to your branch)

File: kitsune/sumo/static/sumo/js/libs/jquery.ajaxupload.js [hunk 1]
```diff
@@ -9,137 +9,138 @@
  * if you want to avoid that).
  */
 
-import dialogSet from "sumo/js/upload-dialog";
```
Comment (akatsoulas) [inline kitsune/sumo/static/sumo/js/libs/jquery.ajaxupload.js:12]: What are the changes in this file?

**Review by akatsoulas — APPROVED** [review]
My proposal here would be to remove the changes to bootstrap and setting.py and merge all the commits into one. Changes that are only coming from linters or formatting tools should be excluded too (is this the case with jquery.ajaxupload.js?). Once we have all the changes into a single commit we can merge it into main since we do need these updates. We can open new PRs for any further changes. r+wc

File: kitsune/settings.py [hunk 2]
```diff
@@ -946,7 +946,7 @@ def JINJA_CONFIG():
 API_LIMIT_PER_PAGE = 0
 
 # Change the default for XFrameOptionsMiddleware.
-X_FRAME_OPTIONS = "DENY"
+X_FRAME_OPTIONS = "SAMEORIGIN"
```
[the remark this answers is not in the document]
Comment (smithellis) [reply kitsune/settings.py:949]: When troubleshooting the KB article editing issue, I was getting console errors about the iframe the js implements to post to, and this resolved it. If this change is concerning I can revert and look at other options.

File: bin/run-bootstrap.sh [hunk 3]
```diff
@@ -3,6 +3,6 @@
 set -ex
 
 # install Node dependencies
-npm run development && npm run webpack:build
+npm run development && npm run webpack:build
```
[the remark this answers is not in the document]
Comment (smithellis) [reply bin/run-bootstrap.sh]: I believe I have removed it.

File: kitsune/settings.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (smithellis) [reply kitsune/settings.py:949]: Should I revert?

File: kitsune/sumo/static/sumo/js/libs/jquery.ajaxupload.js (hunk 1 above)
Comment (smithellis) [reply kitsune/sumo/static/sumo/js/libs/jquery.ajaxupload.js:12]: Pretty sure I deleted and replaced the file content after filling it full of debug/log code that I commited then corrected. So...just space I believe.

File: kitsune/settings.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (akatsoulas) [reply kitsune/settings.py:949]: Yes, let's capture this in a different issue, outside of the jquery upgrade. We may end up doing exactly that but let's focus on it separately

File: kitsune/sumo/static/sumo/js/libs/jquery.ajaxupload.js (hunk 1 above)
Comment (smithellis) [reply kitsune/sumo/static/sumo/js/libs/jquery.ajaxupload.js:12]: The only pertinent change is from jquery's json parser to the now standard issue JSON.parse

**Comment by akatsoulas** [comment]
closing in favor of #5211 . If I missed anything in that PR feel free to open a new one

**Pull request closed without merging**